rinja_parser/src/expr.rs

```diff
@@ -190,26 +191,42 @@ impl<'a> Expr<'a> {
     expr_prec_layer!(band, shifts, token_bitand);
     expr_prec_layer!(shifts, addsub, alt((tag(">>"), tag("<<"))));
     expr_prec_layer!(addsub, muldivmod, alt((tag("+"), tag("-"))));
-    expr_prec_layer!(muldivmod, is_defined, alt((tag("*"), tag("/"), tag("%"))));
+    expr_prec_layer!(muldivmod, is_as, alt((tag("*"), tag("/"), tag("%"))));
 
-    fn is_defined(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
+    fn is_as(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
         let start = i;
         let (i, lhs) = Self::filtered(i, level)?;
-        let (i, rhs) = opt(preceded(
-            ws(keyword("is")),
-            opt(terminated(opt(keyword("not")), ws(keyword("defined")))),
-        ))(i)?;
-        let is_neg = match rhs {
-            None => return Ok((i, lhs)),
-            Some(None) => {
+        let (j, rhs) = opt(ws(identifier))(i)?;
+        let i = match rhs {
+            Some("is") => j,
+            Some("as") => {
+                let (i, target) = opt(identifier)(j)?;
+                let target = target.unwrap_or_default();
+                return match target {
+                    "bool" | "f32" | "f64" | "i8" | "i16" | "i32" | "i64" | "i128" | "isize"
```

Comment:
Missing f16 and f128. Would be nice to have the full list of primitives somewhere centralized because we use it in different places.

Reply:
Good idea! Rust is even smart enough to overlay lists `&["a"]`, `&["a", "b"]` and `&["a", "b", "c"]`, so they all point to the same memory address, but with different lengths.